File: config/Global.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Global 对应 config/Global.py。
type Global struct {
	*session.ApiGroup
}

// NewGlobal 创建 Global API 分组。
func NewGlobal(s *session.Session) *Global {
	return &Global{ApiGroup: session.NewApiGroup(s)}
}

// Languagelist 对应 global/languagelist.xml。
func (g *Global) Languagelist() (map[string]interface{}, error) {
	return g.S.Get("global/languagelist.xml", nil, "config")
}

// Config 对应 global/config.xml。
func (g *Global) Config() (map[string]interface{}, error) {
	return g.S.Get("global/config.xml", nil, "config")
}

// NetType 对应 global/net-type.xml。
func (g *Global) NetType() (map[string]interface{}, error) {
	return g.S.Get("global/net-type.xml", nil, "config")
}

File: src/config/global.rs
//! Global config group (`config/Global.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Global config group.
pub struct GlobalConfig<'a> {
    conn: &'a Connection,
}

impl<'a> GlobalConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        GlobalConfig { conn }
    }

    /// `global/languagelist.xml`. Supported language list.
    pub fn languagelist(&self) -> Result<Value> {
        get_value(self.conn, "config/global/languagelist.xml")
    }

    /// `global/config.xml`. Global configuration.
    pub fn config(&self) -> Result<Value> {
        get_value(self.conn, "config/global/config.xml")
    }

    /// `global/net-type.xml`. Network type.
    pub fn net_type(&self) -> Result<Value> {
        get_value(self.conn, "config/global/net-type.xml")
    }
}
